pkg/apis/core/v1alpha1/liveupdate_types.go

```diff
@@ -53,6 +53,67 @@ type LiveUpdateList struct {
 
 // LiveUpdateSpec defines the desired state of LiveUpdate
 type LiveUpdateSpec struct {
+	// An absolute local path that serves as the basis for all
+	// path calculations.
+	//
+	// Relative paths in this object are calculated relative to the base path. It
+	// cannot be empty.
+	//
+	// +tilt:local-path=true
+	BasePath string `json:"basePath" protobuf:"bytes,1,opt,name=basePath"`
+
+	// Name of the FileWatch object to watch for a list of files that
+	// have recently been updated.
+	//
+	// Every live update must be associated with a FileWatch object
+	// to trigger the update.
+	FileWatchName string `json:"fileWatchName" protobuf:"bytes,2,opt,name=fileWatchName"`
+
+	// Name of the KubernetesDiscovery object to watch for a list of pods
+	// that we're able to update.
+	//
+	// Every live update must be associated with some object for finding
+	// containers. In the future, we expect there to be other types
+	// of container discovery objects (like Docker Compose container discovery).
+	//
+	// +optional
+	KubernetesDiscoveryName string `json:"kubernetesDiscoveryName,omitempty" protobuf:"bytes,3,opt,name=kubernetesDiscoveryName"`
+
+	// A list of relative paths that will immediately stop the live-update for the
+	// current container.
+	//
+	// Used to detect file changes that invalidate the entire container image,
+	// forcing a complete rebuild.
```

Comment:
I don't have a good sense how this will fit into the flow or what "stop the live-update" will mean in practice, would something like this be accurate?

```
A list of relative paths that will cause this live-update to pause until the container has been redeployed.
```

(also the existing wording makes it easy to think the liveupdate does the complete rebuild)

If that's accurate, maybe "container has been redeployed" is a weird condition and we want the PodTemplateSpecHash somewhere?

Reply:
this exists in the model today: https://docs.tilt.dev/api.html#api.fall_back_on

Reply:
(More generally, I'm curious how the reconciler for this will work, both short-term and long-term)